uwblueprint/community-fridge-kw
# feat: recurring delete donation modal

## Brief description. What is this change? 

### [Create delete recurring or delete individual donation modal](https://www.notion.so/uwblueprintexecs/Create-delete-recurring-or-delete-individual-donation-modal-24f8a70dfa5048c19f3d3fe69869538a)

Just noticed that this is PR #100 :) 


## Implementation description. How did you make this change?
* Created separate recurring donation modal 
* Depending on whether the recurringDonationId was null I showed differrent modals
* Delete depended on whether they only wanted to cancel one event or all recurring events
* Toast was added after successful implementation
* Also added a backend change to address the possibility of adding both id and recurringDonationId from my last ticket [here](https://github.com/uwblueprint/community-fridge-kw/pull/93)



## Steps to test
1. Create one one-time donation and one recurring donation
2. Delete the one-time donation and make sure it is actually removed and the toast is showing
3. Delete the recurring donation but only the 'this event' and make sure it is actually removed and the toast is showing
4. Delete the remaining recurring donation events and ensure everything past the day you chose is removed and the toast is showing

## Checklist
